admin/publish-release.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+set -e
+
+# Publish to BinTray if the HEAD commit is tagged with a version number.
+if [ "$PULL_REQUEST" != "false" ]; then
+  echo "Shippable is building a pull request, not publishing."
+  echo "PULL_REQUEST is equal to $PULL_REQUEST"
+  exit 0
+fi
+
+if [ "$BRANCH" != "master" ]; then
+  echo "Shippable is building on branch $BRANCH, not publishing."
+  echo "BRANCH is equal to $BRANCH"
+  exit 0
+fi
+
+numParents=`git log --pretty=%P -n 1 | wc -w | xargs`
+if [ $numParents -ne 2 ]; then
+  echo "$numParents parent commits of HEAD when exactly 2 expected, not publishing."
+  exit 0
+fi
+
+# Shippable only runs a single build when a pull request is merged, so we need to list
+# all tags from the merged commits.
+firstMergedCommit=`git rev-list HEAD^2 --not HEAD^1 | tail -n 1`
```

Comment:
I can't get the `^2` revision working reliably on my git branches. It may not always point to something. I'm actually having a hard time understanding what it even means.

I *think* that `git rev-list HEAD~1..HEAD` is what you want (note that `HEAD~1..HEAD` is the same as `HEAD --not HEAD~1`, but is a little more readable).

FWIW you could add `--reverse --max-count=1` to get the `tail -n 1` effect.